xhr/headers-normalize-response.htm

```diff
@@ -34,9 +36,12 @@ <h1>Whitespace and null in response header values</h1>
 matchHeaderValue("\thello world\t");
 matchHeaderValue("hello      world");
 matchHeaderValue("hello\tworld");
-matchHeaderValue("\0");
+error("\0");
 matchHeaderValue("   ");
 matchHeaderValue("\t");
 matchHeaderValue("");
 
+promise_test(t => {
+  return promise_rejects(t, new TypeError(), fetch("resources/parse-headers.py?my-custom-header="+encodeURIComponent("x\0x")));
+}, "Ensure fetch() rejects too")
```

Comment:
This seems pretty out of place here; another separate file would be much better, in case people are testing fetch() and XHR in isolation.

Reply:
Why would they do that?

Reply:
Because they're working on their fetch() or XHR implementation?

Reply:
You need to support both and they both need to fail in the same way due to both operating on the same low-level primitive.

Reply:
(And it's not like both are tested in the same test so it'd be easy to ignore some results.)

Reply:
(I actually think we might want to merge these directories at some point due to the worry of missing test coverage in either API.)

Reply:
I'm still not comfortable testing fetch in the XHR directory, sorry. It makes maintainers lives harder when they don't know where a given API's tests are.

Reply:
That's already the case though for many cross-purpose tests. I guess we could move this into the `http/` directory...